homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,524 @@
+    OPERATION_MODE_COOL,
+    OPERATION_MODE_HEAT,
+    OPERATION_MODE_OFF,
+    OPERATION_MODES,
+    SYSTEM_STATUS_COOL,
+    SYSTEM_STATUS_HEAT,
+    SYSTEM_STATUS_IDLE,
+    UNIT_FAHRENHEIT,
+)
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_AUX_HEAT,
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_HUMIDITY,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    ATTR_TARGET_TEMP_STEP,
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_AUX_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    STATE_OFF,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_AIRCLEANER_MODE,
+    ATTR_DEHUMIDIFY_SETPOINT,
+    ATTR_DEHUMIDIFY_SUPPORTED,
+    ATTR_HOLD_MODES,
+    ATTR_HUMIDIFY_SETPOINT,
+    ATTR_HUMIDIFY_SUPPORTED,
+    ATTR_SETPOINT_STATUS,
+    ATTR_THERMOSTAT_ID,
+    ATTR_ZONE_ID,
+    ATTR_ZONE_STATUS,
+    ATTRIBUTION,
+    DATA_NEXIA,
+    DOMAIN,
+    MANUFACTURER,
+    NEXIA_DEVICE,
+    UPDATE_COORDINATOR,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_SET_AIRCLEANER_MODE = "set_aircleaner_mode"
+SERVICE_SET_HUMIDIFY_SETPOINT = "set_humidify_setpoint"
+
+SET_FAN_MIN_ON_TIME_SCHEMA = vol.Schema(
+    {
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_AIRCLEANER_MODE): cv.string,
+    }
+)
+
+SET_HUMIDITY_SCHEMA = vol.Schema(
+    {
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_HUMIDITY): vol.All(
+            vol.Coerce(int), vol.Range(min=35, max=65)
+        ),
+    }
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up climate for a Nexia device."""
+
+    nexia_data = hass.data[DOMAIN][config_entry.entry_id][DATA_NEXIA]
+    nexia_home = nexia_data[NEXIA_DEVICE]
+    coordinator = nexia_data[UPDATE_COORDINATOR]
+
+    entities = []
+    for thermostat_id in nexia_home.get_thermostat_ids():
+        thermostat = nexia_home.get_thermostat_by_id(thermostat_id)
+        for zone_id in thermostat.get_zone_ids():
+            zone = thermostat.get_zone_by_id(zone_id)
+            entities.append(NexiaZone(coordinator, zone))
+
+    def humidify_set_service(service):
+        entity_id = service.data.get(ATTR_ENTITY_ID)
+        humidity = service.data.get(ATTR_HUMIDITY)
+        target_zones = []
+
+        for zone in entities:
```

Comment:
How will this work with multiple config entries ?

Reply:
(I don't think actually that `platform.async_register_entity_service` handles it…)

Reply:
Its going to get overwritten.  Only thing I can think to do is create these in async_setup and then use 
```
   entity_registry = await hass.helpers.entity_registry.async_get_registry()
```

Unless there is a better way

Reply:
Actually, this PR is already big enough. I'll just nuke the service for now and figure out how to add it back in a followup

Reply:
services deleted from this pr